.github/actions/k-bench/action.yml

```diff
@@ -0,0 +1,105 @@
+        KUBECONFIG=$PWD/constellation-admin.conf kubectl wait -n kube-system deployments csi-azuredisk-controller --for condition=available --timeout=300s
+        KUBECONFIG=$PWD/constellation-admin.conf kubectl apply -f .github/actions/k-bench/azure_sc.yml
+    - name: Run K-Bench
+      shell: bash
+      working-directory: k-bench
+      run: |
+        mkdir -p ./out
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl create namespace kbench-pod-namespace --dry-run=client -o yaml | KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f -
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f ../.github/actions/k-bench/encrypted_storage.yml
+        KUBECONFIG=$PWD/../constellation-admin.conf ./run.sh -r "kbench-constellation-${{ inputs.cloudProvider }}" -t "default" -o "./out/"
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl delete namespace kbench-pod-namespace --wait=true || true
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl create namespace kbench-pod-namespace --dry-run=client -o yaml | KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f -
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f ../.github/actions/k-bench/encrypted_storage.yml
+        KUBECONFIG=$PWD/../constellation-admin.conf ./run.sh -r "kbench-constellation-${{ inputs.cloudProvider }}" -t "dp_fio" -o "./out/"
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl delete namespace kbench-pod-namespace --wait=true || true
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl create namespace kbench-pod-namespace --dry-run=client -o yaml | KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f -
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f ../.github/actions/k-bench/encrypted_storage.yml
+        KUBECONFIG=$PWD/../constellation-admin.conf ./run.sh -r "kbench-constellation-${{ inputs.cloudProvider }}" -t "dp_network_internode" -o "./out/"
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl delete namespace kbench-pod-namespace --wait=true || true
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl create namespace kbench-pod-namespace --dry-run=client -o yaml | KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f -
+        KUBECONFIG=$PWD/../constellation-admin.conf kubectl apply -f ../.github/actions/k-bench/encrypted_storage.yml
+        KUBECONFIG=$PWD/../constellation-admin.conf ./run.sh -r "kbench-constellation-${{ inputs.cloudProvider }}" -t "dp_network_intranode" -o "./out/"
+    - name: Delete namespace
+      shell: bash
+      run: KUBECONFIG=$PWD/constellation-admin.conf kubectl delete namespace kbench-pod-namespace --wait=true || true
+    - name: Delete PVCs
+      shell: bash
+      run: KUBECONFIG=$PWD/constellation-admin.conf kubectl delete pvc --all --wait=true || true; sleep 15
```

Comment:
I included the wait flag for every namespace deletion to be sure it is deleted before moving on. While arguably here it does not really matter, it does in the case when K-Bench is launched right after it. Since sometimes the namespace might  not yet been deleted, K-Bench will fail to create a new one. Therefore, I found it best to just wait for deletion before every run.

The arbitrary sleep is there to allow enough time to propagate the deletion from pvc -> pv -> GCP disk. I would like to keep it just to be safe and not leave dangling GCP disks around.

Reply:
I see, that makes sense. 

But wouldn't you be able to replace the sleep with some `kubectl wait --for=delete` command (if that works for PVs)?

Because if PV deletion is fast, it should be faster than sleep 15, and in case it's slow it should be waiting until it is eventually deleted. From experience sometimes the CSP API can crap out temporarily and then we are back at the dangling resources problem. 

Or we should have a general cleanup for dangling resources for GCP in general. For Azure we clean up the whole resource group so that shouldn't be an issue there at least (though deletion by K8s would ideally work and should be tested). But that isn't something I would suggest for this PR.

Reply:
Totally understand your point.

I have tracked the lifecycle of a Disk in GCP, `pv`, and `pvc` in the cluster while deleting the namespace. And indeed, the `pv` and the GCP disk stay around for a little longer (like 10 seconds or so) after the *namespace* deletion has finished and the wait is over.

If there was an immediate destruction of the cluster at that point, I don't know whether that could stop the propagation.

As a result, I propose to query the number of disks still present in the namespace after deletion and only continue after they are no longer in the cluster, assuming GCP will handle the rest on their own internally.